### PR TITLE
CLOUDP-118221: [CLI] We should not parse the access token when programmatic keys are available

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-118221

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongocli/blob/master/CONTRIBUTING.md) (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](https://github.com/mongodb/mongocli/blob/master/e2e/E2E-TESTS.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

## Further comments

This PR updates `WithAuthentication` to not parse the access token when programmatic keys are available

### CODE REVIEW

File: internal/store/store.go [hunk 1]
```diff
@@ -219,11 +219,14 @@ func WithAuthentication(c CredentialsGetter) Option {
...
-		t, err := c.Token()
-		if err != nil {
-			return err
+
+		if s.username == "" && s.password == "" {
+			t, err := c.Token()
+			if err != nil {
+				return err
+			}
+			s.accessToken = t
```
Comment (andreaangiolillo) [inline internal/store/store.go:228]: we use the access token only when programmatic keys are not set

**Comment by andreaangiolillo** [comment]
Moving this to WIP: I have discussed this proposal with @johansteyn and it may make sense to keep the access token even if the user is using the programmatic keys for telemetry. Waiting to hear back from the product team to see what to do here

**Review by fmenezes — DISMISSED** [review]
Go code LGTM

File: internal/store/store.go (hunk 1 above)
Comment (andreaangiolillo) [reply internal/store/store.go:228]: yes, correct. I am just waiting to see if I should remove the logic that remove the auth tokens in the config commands

File: internal/cli/atlas/config/init.go [hunk 2]
```diff
@@ -97,6 +99,20 @@ Enter [?] on any option to get help.
...
+func deleteAccessAndRefreshTokenFromProfile() {
+	accessToken := config.GetString(config.AccessTokenField)
+	refreshToken := config.GetString(config.RefreshTokenField)
+
+	if accessToken == "" && refreshToken == "" {
+		return
+	}
+
+	config.Set(config.AccessTokenField, "")
+	config.Set(config.RefreshTokenField, "")
```
Comment (gssbzn) [inline internal/cli/atlas/config/init.go:111]: it'd probably be better to implement a `Unset`

File: internal/store/store.go (hunk 1 above)
Comment (andreaangiolillo) [reply internal/store/store.go:228]: This is not the behavior that I see though. If I have an invalid access token and the programmatic key I get this error without this change. I believe `t, err := c.Token()` may the root cause here

```
./bin/atlas project ls -P default-access-token
Error: token contains an invalid number of segments
```

Comment (gssbzn) [reply internal/store/store.go:228]: well that may be your own fault, `invalid number of segments` means you manually modified that token and the only case I would expect `c.Token()` to error, if you give it a random string and not a valid JWT, and by valid in the generic is a JWT token kind of way, you can verify your string in this page https://jwt.io/, if that page can parse then this function can as well

Comment (andreaangiolillo) [reply internal/store/store.go:228]: I see. I don't think we should parse the access token if we are not using it though

Comment (gssbzn) [reply internal/store/store.go:228]: and I don't think users should be manually editing their tokens, in general my comment in this particular change is not blocking, just saying there's not much gain but it doesn't hurt, deleting things without previous warning does hurt and a lot

**Review by gssbzn — APPROVED** [review]
LGTM

**Pull request merged**